File: problem/0145_Binary Tree Postorder Traversal.js
// Given the root of a binary tree, return the postorder traversal of its nodes' values.

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */

// postOrder: node.left -> node.right -> node
var postorderTraversal = function(root) {
  const result = [];
  const postorder = (node) => {
    if(!node) return;
    let L = postorder(node.left);
    let R = postorder(node.right);
    result.push(node.val);
  }
  postorder(root);
  return result;
};

File: c#-solution/0145. Binary Tree Postorder Traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution
{
  public IList<int> PostorderTraversal(TreeNode root)
  {
    Stack<int> ans = new Stack<int>();
    Stack<TreeNode> stack = new Stack<TreeNode>();
    TreeNode node = root;

    if (node != null) stack.Push(node);

    while (stack.Count != 0)
    {
      node = stack.Pop();
      if (node != null)
      {
        ans.Push(node.val);
        stack.Push(node.left);
        stack.Push(node.right);
      }
    }
    return ans.ToList();

  }
}